security/voters.rst

```diff
@@ -263,106 +263,17 @@ your voter will be executed and you can control access.
-                return true;
-            }
-
-            // ... all the normal voter logic
-        }
-    }
-
-Next, update ``services.yml`` to inject the ``security.access.decision_manager``
-service:
-
-.. configuration-block::
-
-    .. code-block:: yaml
-
-        # app/config/services.yml
-        services:
-            app.post_voter:
-                class: AppBundle\Security\PostVoter
-                arguments: ['@security.access.decision_manager']
-                public: false
-                tags:
-                    - { name: security.voter }
-
-    .. code-block:: xml
-
-        <!-- app/config/services.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                http://symfony.com/schema/dic/services/services-1.0.xsd">
-
-            <services>
-                <service id="app.post_voter"
-                    class="AppBundle\Security\PostVoter"
-                    public="false"
-                >
-                    <argument type="service" id="security.access.decision_manager"/>
-
-                    <tag name="security.voter" />
-                </service>
-            </services>
-        </container>
-
-    .. code-block:: php
-
-        // app/config/services.php
-        use AppBundle\Security\PostVoter;
-        use Symfony\Component\DependencyInjection\Definition;
-        use Symfony\Component\DependencyInjection\Reference;
-
-        $container->register('app.post_voter', PostVoter::class)
-            ->addArgument(new Reference('security.access.decision_manager'))
-            ->setPublic(false)
-            ->addTag('security.voter')
-        ;
-
-That's it! Calling ``decide()`` on the ``AccessDecisionManager`` is essentially
-the same as calling ``isGranted()`` from a controller or other places 
-(it's just a little lower-level, which is necessary for a voter).
-
-.. note::
-
-    The ``security.access.decision_manager`` is private. This means you can't access
-    it directly from a controller: you can only inject it into other services. That's
-    ok: use ``security.authorization_checker`` instead in all cases except for voters.
+to see if the current user has ``ROLE_SUPER_ADMIN``. The ``isGranted()`` method
+is provided by the ``security.authorization_checker`` service via the
+:class:`Symfony\\Component\\Security\\Core\\Authorization\\AccessDecisionManager`
+class. However, you can't inject that service because it causes a
+``CircularReferenceException``.
+
+The solution is to inject the ``service_container`` service and use it to get
+the ``security.authorization_checker`` service and call to the ``isGranted()``
+method. Another solution is to upgrade your Symfony version, because this issue
+was fixed starting from Symfony 3.1.
```

Comment:
The linked ticket is mentioning the debug access decision manager introduced in 3.1 but how could this fix the problem of circular dependency for voters requiring the manager? Are we sure this is fixed? Thanks!